tree.go

```diff
@@ -22,33 +22,12 @@ type Tree interface {
 	Root() ([]byte, error)
 }
 
-var _ Tree = &DefaultTree{}
+// trees is a global map used for keeping track of registered tree constructors for testing and JSON unmarshalling
```

Comment:
[nit] if it's not just used for testing, proposal to remove that from the description

[nit] I think the godoc should recommend a naming structure for custom defined trees. For example:

```suggestion
// trees is a global map used for keeping track of registered tree constructors for JSON serialization.
// The keys of this map should be kebab cased. E.g. "default-tree"
```